### PR TITLE
FUSETOOLS-2958: New UI test - Try to deploy all available templates to a runtime

### PR DESCRIPTION
**Pull request by djelinek**
Signed-off-by: Dominik Jelinek <djelinek@redhat.com>

# Pull Request Checklist

After this checklist is all checked or PR provides explanations for possible pass-through, please put the label "Ready for review" on the PR.

## General

- [ ] Did you use the Jira Issue number in the commit comments?
- [ ] Did you set meaningful commit comments on each commit?
- [ ] Did you sign off all commits for this PR? (git commit -s -m "jira issue number - your commit comment")

## Functional

- [ ] Did the CI job report a successful build?
- [ ] Is the non-happy path working, too?

## Maintainability

- [ ] Are all Sonar reported issues fixed or can they be ignored?
- [ ] Is there no duplicated code?
- [ ] Are method-/class-/variable-names meaningful?

## Tests

- [ ] Are there unit-tests?
- [ ] Are there integration tests (or at least a jira to tackle these)?
- [ ] Do we need a new UI test?

## Legal

- [ ] Have you used the correct file header copyright comment?
- [ ] Have you used the correct year in the headers of new files?

### CODE REVIEW

**Comment by tsedmik** [comment]
testPR

File: uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/DeploymentTest.java [hunk 1]
```diff
@@ -11,96 +11,192 @@
...
 	public static final String BROWSER_URL = "http://localhost:8080/camel-test-spring";
 	public static final String BROWSER_CONTENT_DEPLOYED = "Hello null";
 	public static final String BROWSER_CONTENT_UNDEPLOYED = "404";
 
+	private FuseProjectDefinition project;
+
 	@InjectRequirement
 	private static FuseRequirement serverRequirement;
 
 	@RequirementRestriction
```
Comment (tsedmik) [inline uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/DeploymentTest.java]: Do we need this restriction anymore? IMHO no.

File: uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/DeploymentTest.java [hunk 2]
```diff
@@ -11,96 +11,192 @@
...
+	}
 
-		new WorkbenchShell().setFocus();
-		FuseServerManipulator.removeAllModules(serverRequirement.getConfiguration().getServer().getName());
-		FuseServerManipulator.stopServer(serverRequirement.getConfiguration().getServer().getName());
-		FuseServerManipulator.removeServer(serverRequirement.getConfiguration().getServer().getName());
+	@AfterClass
+	public static void cleanServers() {
+		String serverName = serverRequirement.getConfiguration().getServer().getName();
+		FuseServerManipulator.removeAllModules(serverName);
```
Comment (tsedmik) [inline uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/DeploymentTest.java]: we should use _FuseServerManipulator.removeAllModules(serverName);_ method also in _@After_ method. If something went wrong with the test, it ensures that the project will be undeployed and do not affect next test.

File: uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/DeploymentTest.java [hunk 3]
```diff
@@ -142,49 +238,110 @@ public void setupManageServers() {
...
+		}
+
+		// deploy project (add module...)
+		FuseServerManipulator.addModule(serverName, PROJECT_NAME);
+		assertTrue(FuseServerManipulator.hasServerModule(serverName, PROJECT_NAME));
+
+		// check deployed
+		switch (RUNTIME_TYPE) {
+		case "Karaf":
+			assertKaraf("The project was not properly deployed!", "deploy");
```
Comment (tsedmik) [inline uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/DeploymentTest.java]: what about using Enum for "deploy" and "undeploy" values?

**Comment by djelinek** [comment]
@tsedmik I implemented your requested changes into PR. Thank you :-)

**Pull request merged**